### PR TITLE
Combined Pull Requests 

### PR DESCRIPTION
**Pull request by divyansh-sh**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -2345,7 +2366,7 @@ private boolean isIntervalHourAndSecond(SqlCall call) {
       case DOUBLE:
         return createSqlDataTypeSpecByName("FLOAT64", typeName);
       case REAL:
-        return createSqlDataTypeSpecByName("FLOAT32", typeName);
+        return createSqlDataTypeSpecByName("BIGNUMERIC", typeName);
```
Comment (rushikeshsapate-dm) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:2369]: REAL = 32-bit floating-point 
why are we changing this to BIGNUMERIC?

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 2]
```diff
@@ -8296,50 +8321,6 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedSparkQuery));
   }
 
-  @Test public void testQualifyWithReferenceFromSelect() {
```
Comment (rushikeshsapate-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: Could you explain the reason for removing these test cases?

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 3]
```diff
@@ -11572,7 +11548,7 @@ private static RelNode toLogical(RelNode rel, RelBuilder builder) {
         .project(builder.field("DNAME"), builder.field("DEPTNO"))
         .build();
 
-    final ViewChildProjectRelTrait projectViewTrait = new ViewChildProjectRelTrait(true, false);
+    final ViewChildProjectRelTrait projectViewTrait = new ViewChildProjectRelTrait(true);
```
Comment (rushikeshsapate-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: kindly check this

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 4]
```diff
@@ -3889,6 +3897,22 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
           OperandTypes.family(SqlTypeFamily.DATETIME, SqlTypeFamily.STRING),
           SqlFunctionCategory.SYSTEM);
 
+  @LibraryOperator(libraries = {TERADATA})
+  public static final SqlAggFunction REGR_AVGX =
```
[the remark this answers is not in the document]
Comment (divyansh-sh) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:3909]: Added test case

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 2 above)
Comment (divyansh-sh) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: Resolved

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 3 above)
Comment (divyansh-sh) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: Resolved

**Comment by rushikeshsapate-dm** [comment]
LGTM

**Review by dm-shantanuc — APPROVED** [review]

**Pull request merged**